### DISCUSSION TITLE
Batch operator boolean only takes one type.

### DISCUSSION THREAD

**Discussion by fire** [elalish/manifold#1095] [General]
Is it possible to add a tuple of mesh and operator for batching?

**Comment by fire** [comment]
https://manifoldcad.org/docs/html/classmanifold_1_1_manifold.html#a1b30c42e224bda7e71a3ed3cf02319c1

**Comment by elalish** [comment]
I don't understand what you mean. Are you hoping for something like `BatchBoolean({{a, union}, {b, difference}, {c, intersection}})`? If so, then no, that would be nonsensical, as the ordering (grouping) of the operations would have a huge effect on the result.

**Comment by elalish** [comment]
Anyway, we already reorder your Booleans and batch them automatically to maximize performance - this is why we delay evaluation as long as we can. So as long as you don't stop and ask for status between operations, you should be getting it sped up as much as can be.

**Comment by fire** [comment]
We check for the status of operations in input but not between CSG operations. 

```C++
r_manifold = manifold::Manifold(mesh);
manifold::Manifold::Error error = r_manifold.Status();
if (error == manifold::Manifold::Error::NoError) {
    return;
}
```

**Comment by pca006132** [comment]
The idea is that you should not do this, it suffices to just check the status of the final operation. We will propagate it.
